error LNK2001: unresolved external symbol "public: __cdecl log4cplus::CallbackAppender::CallbackAppender

Describe your problem here.

1. Pull the code from:
SHA-1: fe3d648e1bc553b51b752707000ff74fb9bf4969
* Merge pull request #450 from solicomo/=-in-sub-ini-file-path
Support "=" in sub-config file path

2. Use VS2019 16.4 to open the solution, update the whole project to VS2019

3. Change the C++ language to ISO C++17 Standard (std:c++17) in VS2019

4. compile the static lib "log4cplusS"

5. after I use it in an vs2019 test demo, I have this error:
error LNK2001: unresolved external symbol "public: __cdecl log4cplus::CallbackAppender::CallbackAppender(void (__cdecl*)(void *,wchar_t const *,wchar_t const *,int,wchar_t const *,wchar_t const *,unsigned __int64,unsigned long,wchar_t const *,wchar_t const *,int),void *)" 
(??0CallbackAppender@log4cplus@@QEAA@P6AXPEAXPEB_W1H11_KK11H@Z0@Z)

The code of 2.0.4 works good without this issue. Could you help me to solve it so I can update my project. Thank you very much.


Also, provide the following information:

 - version: the code from:
SHA-1: fe3d648e1bc553b51b752707000ff74fb9bf4969
* Merge pull request #450 from solicomo/=-in-sub-ini-file-path
Support "=" in sub-config file path

 - operating system, CPU, bitness: Win10 1909, AMD Ryzen 1950X

 - `configure` script, or CMake, etc., flags and settings: VS2019 16.4 compile the static lib "log4cplusS"
 - flags and settings used by your application: None
 - compiler and its version:  VS2019 16.4

--- a/msvc14/log4cplusS.vcxproj
+++ b/msvc14/log4cplusS.vcxproj
@@ -331,6 +331,7 @@
       <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
+    <ClCompile Include="..\src\callbackappender.cxx" />
     <ClCompile Include="..\src\clogger.cxx" />
     <ClCompile Include="..\src\configurator.cxx">
       <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Debug_Unicode|Win32'">%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
@@ -904,6 +905,7 @@
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\include\log4cplus\appender.h" />
+    <ClInclude Include="..\include\log4cplus\callbackappender.h" />
     <ClInclude Include="..\include\log4cplus\clogger.h" />
     <ClInclude Include="..\include\log4cplus\config.hxx" />
     <ClInclude Include="..\include\log4cplus\configurator.h" />

--- a/msvc14/log4cplusS.vcxproj.filters
+++ b/msvc14/log4cplusS.vcxproj.filters
@@ -187,6 +187,9 @@
     <ClCompile Include="..\src\connectorthread.cxx">
       <Filter>helpers</Filter>
     </ClCompile>
+    <ClCompile Include="..\src\callbackappender.cxx">
+      <Filter>Appenders</Filter>
+    </ClCompile>
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\include\log4cplus\appender.h">
@@ -386,6 +389,9 @@
     </ClInclude>
     <ClInclude Include="..\include\log4cplus\thread\impl\syncprims-cxx11.h">
       <Filter>thread\impl</Filter>
+    </ClInclude>
+    <ClInclude Include="..\include\log4cplus\callbackappender.h">
+      <Filter>Appenders</Filter>
     </ClInclude>
   </ItemGroup>
   <ItemGroup>

